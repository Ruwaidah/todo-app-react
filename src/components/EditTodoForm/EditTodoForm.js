import { useRef, useState } from "react";
import "./EditTodoForm.css";

const EditTodoForm = (props) => {
  const editInput = useRef();
  const [value, setValue] = useState(props.todo);
  const [errorInput, setErrorInput] = useState(false);

  const changeHandler = (e) => {
    setErrorInput(false);
    setValue(e.target.value);
  };

  const submitHandler = (e) => {
    e.preventDefault();
    if (value === "") {
      setErrorInput(true);
      editInput.current.focus();
    } else {
      props.editTask(value, props.index);
      props.setIsEdit(false);
    }
  };
  return (
    <form className="EditTodoForm" onSubmit={submitHandler}>
      <label htmlFor="editTask">Task:</label>
      <input
        className={errorInput ? "error" : null}
        id="editTask"
        name="edit"
        ref={editInput}
        value={value}
        onChange={changeHandler}
      />
      <input type="submit" value="edit" />
      <button type="button" onClick={() => props.setIsEdit(false)}>
        Cancel
      </button>
    </form>
  );
};

export default EditTodoForm;
