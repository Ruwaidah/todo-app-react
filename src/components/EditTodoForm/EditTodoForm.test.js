import { render } from "@testing-library/react";
import EditTodoForm from "./EditTodoForm";

it("it render without crash", () => {
  render(<EditTodoForm />);
});

it("should match snapshot", () => {
  const { asFragment } = render(<EditTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
