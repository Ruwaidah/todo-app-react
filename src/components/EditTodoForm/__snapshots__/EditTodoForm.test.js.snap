// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditTodoForm"
  >
    <label
      for="editTask"
    >
      Task:
    </label>
    <input
      id="editTask"
      name="edit"
      value=""
    />
    <input
      type="submit"
      value="edit"
    />
    <button
      type="button"
    >
      Cancel
    </button>
  </form>
</DocumentFragment>
`;
